Content.Shared/SegmentedEntity/SegmentedEntitySystem.cs

```diff
@@ -0,0 +1,306 @@
+
+                var joint = _jointSystem.CreateDistanceJoint(attachedUid, segmentUid, id: ("Segment" + segment.segment.SegmentNumber + segment.segment.Lamia));
+                joint.CollideConnected = false;
+                joint.Stiffness = 0.2f;
+            }
+            _segments.Clear();
+        }
+        private void OnInit(EntityUid uid, SegmentedEntityComponent component, ComponentInit args)
+        {
+            EnsureComp<PortalExemptComponent>(uid); //Temporary, remove when Portal handling is added
+            Math.Clamp(component.NumberOfSegments, 2, 18);
+            Math.Clamp(component.TaperOffset, 1, component.NumberOfSegments - 1);
+            SpawnSegments(uid, component);
+        }
+
+        private void OnShutdown(EntityUid uid, SegmentedEntityComponent component, ComponentShutdown args)
+        {
+            if (_net.IsClient)
+                return;
+
+            foreach (var segment in component.Segments)
+            {
+                QueueDel(GetEntity(segment));
+            }
+
+            component.Segments.Clear();
+        }
+
+        /// <summary>
+        ///     TODO: Full Self-Test function that intelligently checks the status of where everything is, and calls whatever
+        ///     functions are appropriate
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        public void SegmentSelfTest(EntityUid uid, SegmentedEntityComponent component)
+        {
+
+        }
+
+        /// <summary>
+        ///     TODO: Function that ensures clothing visuals, to be called anytime the tail is reset
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="segment"></param>
+        private void EnsureSnekSock(EntityUid uid, SegmentedEntityComponent segment)
+        {
+
+        }
+        public void OnStoreSnekAttempt(EntityUid uid, SegmentedEntityComponent comp, ref StoreMobInItemContainerAttemptEvent args)
+        {
+            args.Cancelled = true;
+        }
+
+        private void OnJointRemoved(EntityUid uid, SegmentedEntityComponent component, JointRemovedEvent args)
+        {
+            if (!component.Segments.Contains(GetNetEntity(args.OtherEntity)))
+                return;
+
+            DeleteSegments(component);
+        }
+
+        private void DeleteSegments(SegmentedEntityComponent component)
+        {
+            if (_net.IsClient)
+                return; //Client is not allowed to predict QueueDel, it'll throw an error(but won't crash in Release build)
+
+            foreach (var segment in component.Segments)
+                QueueDel(GetEntity(segment));
+
+            component.Segments.Clear();
+        }
+
+        /// <summary>
+        ///     Public call for a SegmentedEntity to reset their tail completely.
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
```

Comment:
either write commentaries explaining what this params are or remove this from xaml docs entirely.